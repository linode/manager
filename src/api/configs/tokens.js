import {
  genConfig, ReducerGenerator, genActions,
  ONE, MANY, DELETE,
} from '~/api/apiResultActionReducerGenerator';

export const config = genConfig({
  plural: 'tokens',
  singular: 'token',
  endpoint: id => `/account/tokens/${id}`,
  supports: [ONE, MANY, DELETE],
});

export const actions = genActions(config);
export const { reducer } = new ReducerGenerator(config);
