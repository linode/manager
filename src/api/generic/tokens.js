import {
  genConfig, ReducerGenerator, genActions,
  ONE, MANY, DELETE, POST, PUT,
} from '~/api/internal';

export const config = genConfig({
  plural: 'tokens',
  endpoint: id => `/profile/tokens/${id}`,
  supports: [ONE, MANY, DELETE, POST, PUT],
});

export const actions = genActions(config);
export const { reducer } = new ReducerGenerator(config);
