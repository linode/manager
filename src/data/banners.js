export const testApiBanner = {
  entity: {
    id: 1234,
    type: 'ticket',
    label: 'testing ticket',
  },
  type: 'important_ticket',
  when: null,
};

export const testBillingBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'Please update your billing information.',
  },
};

export const testAbuseBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'ToS Violation',
  },
  type: 'abuse_ticket',
};

export const testAbuseBanner2 = {
  ...testAbuseBanner,
  entity: {
    id: 1236,
  },
};

export const importantBanner = [
  testApiBanner,
];

export const importantBanners = [
  testApiBanner,
  testBillingBanner,
];

export const abuseBanner = [
  testAbuseBanner,
];

export const abuseBanners = [
  testAbuseBanner,
  testAbuseBanner2,
];

export const banners = [
  testApiBanner,
  testBillingBanner,
  testAbuseBanner,
];
