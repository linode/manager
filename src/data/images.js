export const apiTestImage = {
  id: 'linode/arch2016.05',
  recommended: true,
  vendor: 'Arch',
  label: 'Arch Linux 2016.05',
  size: 550,
  created: '2009-01-01T00:00:00',
  description: 'Test Image',
  type: 'manual',
  created_by: 'linode',
  is_public: true,
  deprecated: false,
};

export const testLinodeImage = {
  ...apiTestImage,
  _polling: false,
};

export const publicImages = {
  'linode/arch2016.05': testLinodeImage,
  'linode/debian7': {
    ...testLinodeImage,
    id: 'linode/debian7',
    vendor: 'Debian',
    label: 'Debian 7',
  },
  'linode/debian8.1': {
    ...testLinodeImage,
    id: 'linode/debian8.1',
    vendor: 'Debian',
    label: 'Debian 8.1',
  },
  'linode/debian6': {
    ...testLinodeImage,
    id: 'linode/debian6',
    recommended: false,
    vendor: 'Debian',
    label: 'Debian 6',
  },
};

export const testPrivateImage = {
  ...testLinodeImage,
  id: 'private/38',
  label: 'testImage',
  description: 'image details',
  vendor: null,
  created: '2017-08-08T13:55:16',
  created_by: 'zues',
  size: 2048,
  type: 'manual',
  is_public: false,
};

export const privateImages = {
  'private/38': testPrivateImage,
  'private/39': {
    ...testPrivateImage,
    id: 'private/39',
    label: 'nextImage',
    description: 'image from deleted disk',
    created: '2017-08-08T13:55:16',
    created_by: 'zues',
    size: 2048,
    type: 'automatic',
    is_public: false,
  },
};

export const images = {
  ...privateImages,
  ...publicImages,
};
