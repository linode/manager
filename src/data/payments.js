export const apiTestPayment = {
  usd: 5.0,
  id: 1000,
  date: '2015-11-12T16:11:59',
};

export const payments = [0, 1, 2, 3, 4].map(i => ({
  ...apiTestPayment,
  id: apiTestPayment.id + i,
  usd: apiTestPayment.usd + i,
}));
