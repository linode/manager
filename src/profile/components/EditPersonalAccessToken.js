import React, { PropTypes, Component } from 'react';

import { Input, ModalFormGroup } from 'linode-components/forms';
import { FormModalBody } from 'linode-components/modals';

import { hideModal, showModal } from '~/actions/modal';
import { tokens } from '~/api';
import { dispatchOrStoreErrors } from '~/api/util';


export default class EditPersonalAccessToken extends Component {
  static title = 'Edit Personal Access Token'

  static trigger(dispatch, token) {
    dispatch(showModal(EditPersonalAccessToken.title, (
      <EditPersonalAccessToken
        id={token.id}
        label={token.label}
        dispatch={dispatch}
        close={() => dispatch(hideModal())}
      />
    )));
  }

  constructor(props) {
    super(props);

    this.state = {
      errors: {},
      label: props.label,
    };
  }

  onChange = ({ target: { name, value } }) => this.setState({ [name]: value })

  onSubmit = () => {
    const { dispatch, id, close } = this.props;
    const { label } = this.state;

    return dispatch(dispatchOrStoreErrors.call(this, [
      () => tokens.put({ label }, id),
      close,
    ]));
  }

  render() {
    const { errors, label } = this.state;

    return (
      <FormModalBody
        onSubmit={this.onSubmit}
        onCancel={this.props.close}
        errors={errors}
        analytics={{ title: EditPersonalAccessToken.title }}
      >
        <ModalFormGroup id="label" label="Label" apiKey="label" errors={errors}>
          <Input
            name="label"
            id="label"
            placeholder="My token"
            value={label}
            onChange={this.onChange}
          />
        </ModalFormGroup>
      </FormModalBody>
    );
  }
}

EditPersonalAccessToken.propTypes = {
  dispatch: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
