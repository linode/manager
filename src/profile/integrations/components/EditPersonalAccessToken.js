import React, { PropTypes, Component } from 'react';

import { CancelButton } from 'linode-components/buttons';
import {
  Form,
  FormSummary,
  Input,
  ModalFormGroup,
  SubmitButton,
} from 'linode-components/forms';

import { tokens } from '~/api';
import { dispatchOrStoreErrors } from '~/api/util';


export default class EditPersonalAccessToken extends Component {
  constructor(props) {
    super(props);

    this.state = {
      errors: {},
      label: props.label,
      loading: false,
    };
  }

  onChange = ({ target: { name, value } }) => this.setState({ [name]: value })

  onSubmit = () => {
    const { dispatch, id, close } = this.props;
    const { label } = this.state;

    return dispatch(dispatchOrStoreErrors.call(this, [
      () => tokens.put({ label }, id),
      close,
    ]));
  }

  render() {
    const { close } = this.props;
    const { errors, label, loading } = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <ModalFormGroup id="label" label="Label" apiKey="label" errors={errors}>
          <Input
            name="label"
            id="label"
            placeholder="My token"
            value={label}
            onChange={this.onChange}
          />
        </ModalFormGroup>
        <div className="Modal-footer">
          <CancelButton onClick={close} />
          <SubmitButton disabled={loading} />
          <FormSummary errors={errors} />
        </div>
      </Form>
    );
  }
}

EditPersonalAccessToken.propTypes = {
  dispatch: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
