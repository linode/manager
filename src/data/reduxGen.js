import { ONE, MANY, DELETE } from '~/api/internal.ts';

import { configsNodeBalancer, noGroupNodeBalancer } from '~/data/nodebalancers';

export const resource = { ...configsNodeBalancer };
export const page = {
  data: [configsNodeBalancer, noGroupNodeBalancer],
  pages: 1,
  results: 1,
  page: 1,
  nodebalancers: [configsNodeBalancer, noGroupNodeBalancer],
};

export const testConfigOne = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [ONE],
};

export const testConfigMany = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [MANY],
};

export const testConfigDelete = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [DELETE],
};
