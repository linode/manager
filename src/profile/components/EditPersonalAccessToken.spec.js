import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import EditPersonalAccessToken from '~/profile/components/EditPersonalAccessToken';

import { changeInput, expectDispatchOrStoreErrors, expectRequest } from '~/test.helpers';
import { testToken } from '~/data/tokens';


describe('profile/components/EditPersonalAccessToken', () => {
  const sandbox = sinon.sandbox.create();

  afterEach(() => {
    sandbox.restore();
  });

  const dispatch = sandbox.stub();

  it('modifies a token', async () => {
    const close = sandbox.spy();
    const page = shallow(
      <EditPersonalAccessToken
        dispatch={dispatch}
        close={close}
        label={testToken.label}
        id={testToken.id}
      />
    );

    changeInput(page, 'label', 'My awesome token');

    await page.props().onSubmit();

    expect(dispatch.callCount).toBe(1);
    await expectDispatchOrStoreErrors(dispatch.firstCall.args[0], [
      ([fn]) => expectRequest(fn, `/profile/tokens/${testToken.id}`, {
        method: 'PUT',
        body: {
          label: 'My awesome token',
        },
      }),
    ], 1);

    expect(close.callCount).toBe(1);
  });
});
