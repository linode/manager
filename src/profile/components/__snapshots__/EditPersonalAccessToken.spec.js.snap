// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile/components/EditPersonalAccessToken should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditPersonalAccessToken
    close={[Function]}
    dispatch={[MockFunction]}
    id={1}
    label="personal"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "analytics": Object {
        "title": "Edit Personal Access Token",
      },
      "buttonDisabledText": "Saving",
      "buttonText": "Save",
      "children": <ModalFormGroup
        apiKey="label"
        errors={Object {}}
        id="label"
        label="Label"
      >
        <Input
          className=""
          id="label"
          name="label"
          onChange={[Function]}
          placeholder="My token"
          value="personal"
        />
      </ModalFormGroup>,
      "errors": Object {},
      "noCancel": false,
      "noCancelEvent": false,
      "noSubmit": false,
      "onCancel": [Function],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "apiKey": "label",
        "children": <Input
          className=""
          id="label"
          name="label"
          onChange={[Function]}
          placeholder="My token"
          value="personal"
        />,
        "errors": Object {},
        "id": "label",
        "label": "Label",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "id": "label",
          "name": "label",
          "onChange": [Function],
          "placeholder": "My token",
          "value": "personal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "analytics": Object {
          "title": "Edit Personal Access Token",
        },
        "buttonDisabledText": "Saving",
        "buttonText": "Save",
        "children": <ModalFormGroup
          apiKey="label"
          errors={Object {}}
          id="label"
          label="Label"
        >
          <Input
            className=""
            id="label"
            name="label"
            onChange={[Function]}
            placeholder="My token"
            value="personal"
          />
        </ModalFormGroup>,
        "errors": Object {},
        "noCancel": false,
        "noCancelEvent": false,
        "noSubmit": false,
        "onCancel": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "apiKey": "label",
          "children": <Input
            className=""
            id="label"
            name="label"
            onChange={[Function]}
            placeholder="My token"
            value="personal"
          />,
          "errors": Object {},
          "id": "label",
          "label": "Label",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "id": "label",
            "name": "label",
            "onChange": [Function],
            "placeholder": "My token",
            "value": "personal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
